Feature user toolbar integration

elad-m opened the pull request:
This has a lot more than the branch name mainly: Deleting and editing product, cool photo shooting tips slider, app Launcher Icon, using the Factory, the HeaderClickListener and most importantly: holding a user session.

elad-m committed:
Redesigned the user profile activity. Added delete functionality of items but  it works for a hardcoded user-id (Aviad's).
Action bar transitions work for both my activities, and written for the others (but don't work since they don't exist yet).
Changed ClothingItem API.
All in all: now we can add items and delete them from a constant user.

elad-m committed:
Redesigned the user profile activity. Added delete functionality of items but  it works for a hardcoded user-id (Aviad's).
Action bar transitions work for both my activities, and written for the others (but don't work since they don't exist yet).
Changed ClothingItem API.
All in all: now we can add items and delete them from a constant user.

elad-m committed:
Added Rotem-approved app launcher icon.
Added some string resources (The ones I did not added are temporary ones
and formatting ones)

elad-m committed:
Added Rotem-approved app launcher icon.
Added some string resources (The ones I did not added are temporary ones
and formatting ones)

elad-m committed:
After rebasing on Pages (Natan's)

elad-m committed:
renaming clothing_item.xml -> clothing_item_layout.xml
fixing bugs after rebasing (style Bridge, ui->UI)
GiveItemActivity - load Properties in order dynamically, no photo/title
 error  red indication, floating wider GIVE button, redesigning title
 text bar

elad-m committed:
Added the three dots for when the item title is too long. Redesigned the
user greeting and compliment to fit

elad-m committed:
Created Basic functionality of Tips in GiveItemActivity (missing button design, dots alignment and a little slow to load)

elad-m committed:
Usage of Factory class.
 UserProfile now calls with EXTRA productId to ItemPageActivity (not sure what the name is).
 Updated the productId from int to long
 Fixed some ui UI issues

elad-m committed:
leftovers of the horrible conflicts when pushing

izipris committed:
handled upload image issue

elad-m committed:
Uploading a photo really works now. Edit works except for image changes..

elad-m committed:
Edit returns after pressing "done" to user profile page
The tips now have been designed with Rotem, but waiting for missing images.
User profile now loads the first name of the user, and when back is pressed
is exiting the app.
Removed the placeholders from the ClothingItemAdapter because it's ugly,
and grey background looks better.
Fixing the Give/Done buttons design and bug with keyboard.
Redesigned dialogs' buttons.
Added attribute to AndroidManifest to solve the keyboard bug.

elad-m committed:
Adding text before title bar, aligning and refactoring
 Adding a line below the give/done button

elad-m committed:
Added missing photos for tips and deleted the bad ones.

elad-m committed:
A few Rotem-inspected design fixes: button line and centering.

elad-m committed:
Fix: Edit now load also the brand

elad-m committed:
Incorporated the use HeaderClickListener in my activities.
Updated use of ProductPage with User and Product as extras.
Getting rid of the infamous ClothingItem.java for a Product, with all changes depending on that.
Ordering GiveItemActivity code so it flows better when reading.

elad-m committed:
User activities are now operating completely on User sent via Extras.
Fixed the failed to login typo since I got it so many times.
Added an attribute for GiveItemActivity in the manifest.
Added OAuth client ID with my SHA-1.

elad-m committed:
Commit after rebase of merge #12: deleting the duplicate getIdAsInt method
from Category.java

elad-m committed:
Fixed a bug: Creating the no photo or title dialog leads to crash. While at it changed the dialogs' corners to be rounder.

izipris commented on app/src/main/java/com/benefit/activities/UserProfileActivity.java:
```diff
@@ -1,91 +1,235 @@
+        long productId = productToDelete.getId();
+        String productTitle = productToDelete.getTitle();
+        AlertDialog.Builder builder = new AlertDialog.Builder(this);
+        builder.setMessage("Are you sure you want to delete \"" + productTitle + "\" ?")
```
Let's put strings that we present to the user in the strings file

elad-m replied:
Alright, working on it.

izipris commented on app/src/main/java/com/benefit/activities/UserProfileActivity.java:
```diff
@@ -1,91 +1,235 @@
+        AlertDialog.Builder builder = new AlertDialog.Builder(this);
+        builder.setMessage("Are you sure you want to delete \"" + productTitle + "\" ?")
+                .setCancelable(false)
+                .setPositiveButton("Yes, I am sure", new DialogInterface.OnClickListener() {
```
strings file

elad-m replied:
Alright, working on it.

izipris commented on app/src/main/java/com/benefit/utilities/staticClasses/Converters.java:
```diff
@@ -33,4 +33,15 @@ public static Boolean isSubstring(String s1,String s2){
         return false;
     }
+
+    public static int getMetaCategoryFromCategoryId(long categoryId) {
```
why do we need this method? the data is available in the category object

elad-m replied:
Because here we hold a product object, and it only holds the category Id as an int, not the object. Is calling the category service with getCategoryById better?

izipris replied:
I think doing it via the categoryService makes more logical sense, because here you defined ids hard-coded, what if we create more categories which are not in this mapping (and we'll do that, I have assignment about it)? It means that for each category that we have, we need to update this mapping also, and it's a violation of the single source principle. So i think if you have the option to read the actual data from categoryService, it's better

elad-m replied:
Done

izipris commented on gradlew:
```diff
@@ -28,7 +44,7 @@ APP_NAME="Gradle"
 
 # Add default JVM options here. You can also use JAVA_OPTS and GRADLE_OPTS to pass JVM options to this script.
-DEFAULT_JVM_OPTS=""
+DEFAULT_JVM_OPTS='"-Xmx64m" "-Xms64m"'
```
is it your local parameters, or mandatory to push to the git?

elad-m replied:
No idea what this is about. I will return it to the previous one.

elad-m committed:
Github Review: Added strings for deletion dialog, returned weird gradlew to previous state, using now LiveData for
getting the parent category when editing a product.
Other: Removed the default greeting (to only load the username-specific greeting)
